src/function/scalar/list/list_resize.cpp

```diff
@@ -0,0 +1,211 @@
+				                                     list_entries[l_index].offset + list_index, result_child_offset);
+			}
+			result_child_offset++;
+		}
+
+		// if the new size is larger than the old size, fill in the default value
+		auto new_size_entry = (idx_t)new_size_entries[new_index];
+		for (idx_t j = values_to_copy; j < new_size_entry; j++) {
+			if (default_vector && default_data.validity.RowIsValid(def_index)) {
+				COPY_FUNCTION::template Operation<T>(result_child_data, default_entries, default_vector, result_child,
+				                                     def_index, result_child_offset);
+			} else {
+				FlatVector::SetNull(result_child, result_child_offset, true);
+			}
+			result_child_offset++;
+		}
+	}
+
+	if (args.AllConstant()) {
+		result.SetVectorType(VectorType::CONSTANT_VECTOR);
+	}
+}
+
+void ListResizeFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	D_ASSERT(args.data[1].GetType().id() == LogicalTypeId::BIGINT);
+	if (result.GetType().id() == LogicalTypeId::SQLNULL) {
+		FlatVector::Validity(result).SetInvalid(0);
+		return;
+	}
+	auto child_type = ListType::GetChildType(args.data[0].GetType()).InternalType();
+	switch (child_type) {
+	case PhysicalType::BOOL:
+	case PhysicalType::INT8:
+		TemplatedListResizeFunction<int8_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::INT16:
+		TemplatedListResizeFunction<int16_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::INT32:
+		TemplatedListResizeFunction<int32_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::INT64:
+		TemplatedListResizeFunction<int64_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::INT128:
+		TemplatedListResizeFunction<hugeint_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::UINT8:
+		TemplatedListResizeFunction<uint8_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::UINT16:
+		TemplatedListResizeFunction<uint16_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::UINT32:
+		TemplatedListResizeFunction<uint32_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::UINT64:
+		TemplatedListResizeFunction<uint64_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::FLOAT:
+		TemplatedListResizeFunction<float, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::DOUBLE:
+		TemplatedListResizeFunction<double, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::VARCHAR:
+		TemplatedListResizeFunction<string_t, StringCopyValue>(args, result);
+		break;
+	case PhysicalType::INTERVAL:
+		TemplatedListResizeFunction<interval_t, NestedCopyValue>(args, result);
+		break;
+	case PhysicalType::STRUCT:
+	case PhysicalType::LIST:
+		TemplatedListResizeFunction<int8_t, NestedCopyValue>(args, result);
+		break;
+	default:
+		throw NotImplementedException("This function has not been implemented for physical type %s",
```

Comment:
Physical types might not be as clear to the user as logical types, so we should tell the user the logical type instead.

Reply:
I actually took this from contains_or_position, should we change it there too so that its consistent?

Reply:
Sounds good to me, yes!